### PR TITLE
[llvm] Support to export dll on windows

### LINKED ISSUE #26325: [llvm] Turn on LLVM_EXPORT_SYMBOLS_FOR_PLUGINS to support custom pass on Windows
Based on this [discussion](https://discourse.llvm.org/t/how-to-create-pass-independently-on-windows/474), it is quite non-trivial to create a custom pass (which is built as a DLL) for LLVM on Windows. A suggested solution is, turn on flag ```LLVM_EXPORT_SYMBOLS_FOR_PLUGINS ``` (and perhaps a few others like ```LLVM_ENABLE_PLUGINS```) in CMake to enable this feature.

The discussion gave a hacky alternative: compile the whole LLVM project into a DLL. I think this approach isn't quite preferable.

A similar explanation can be found [here](https://stackoverflow.com/a/41980033).

### PR DESCRIPTION
**Pull request by FrankXie05**
Fix #26325

Note: Port `llvm` supports export `DLL` on `Windows` by passing option `LLVM_EXPORT_SYMBOLS_FOR_PLUGINS`, and supports `Plugins` correctly.

### CODE REVIEW

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -1254,8 +1254,6 @@ vcpkg-ci-llvm:x64-linux=pass
 vcpkg-ci-llvm:x64-osx=pass
 vcpkg-ci-llvm:x64-windows-static-md=pass
 vcpkg-ci-llvm:x64-windows-static=pass
-vcpkg-ci-llvm:x64-windows=pass
-vcpkg-ci-llvm:x86-windows=pass
```
Comment (Neumann-A) [inline scripts/ci.baseline.txt:1258]: Why is this removed ?

**Comment by MonicaLiu0311** [comment]
Closing this PR since it seems that no progress is being made. Please ping us to reopen if work is still being done.
